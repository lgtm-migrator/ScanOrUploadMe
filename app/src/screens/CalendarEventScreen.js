import React from "react";
import { Button, Text, TextInput, View, StyleSheet, Alert } from "react-native";
import { withNavigation } from "react-navigation";
import { Calendar, Permissions } from "expo";
import { _storeData, _retrieveData } from "../helpers/localStorage";

class CalendarEventScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      id: "",
      summary: "",
      description: "",
      location: "",
      startTime: "",
      endTime: ""
    };
  }

  async componentDidMount() {
    try {
      // TODO: decide what and when an event is fetched
      this._findEvent();
    } catch (error) {
      Alert.alert(error);
      console.log({ error });
    }
  }

  render() {
    return (
      <View>
        <Text style={styles.textContainer}>
          Image URL: {this.props.navigation.getParam("photoUri")}
        </Text>
        <Text style={styles.textContainer}>
          Calendar Event:{" "}
          {"\n" +
            " DESCRIPTION: " +
            this.state.description +
            " SUMMARY: " +
            this.state.summary +
          " LOCATION: " +
          this.state.location +
            " START: " +
            this.state.startTime +
            " END: " +
            this.state.endTime}
        </Text>
        <TextInput
          style={{ height: 40 }}
          placeholder={"SUMMARY"}
          value={this.state.summary}
          onChangeText={text => this.setState({ summary: text })}
        />
        <TextInput
          style={{ height: 40 }}
          placeholder={"DESCRIPTION"}
          value={this.state.description}
          onChangeText={text => this.setState({ description: text })}
        />
        <TextInput
          style={{ height: 40 }}
          placeholder={"LOCATION:"}
          value={this.state.location}
          onChangeText={text => this.setState({ location: text })}
        />
        <TextInput
          style={{ height: 40 }}
          placeholder={"START_TIME"}
          value={this.state.startTime}
          onChangeText={text => this.setState({ startTime: text })}
        />
        <TextInput
          style={{ height: 40 }}
          placeholder={"END_TIME"}
          value={this.state.endTime}
          onChangeText={text => this.setState({ endTime: text })}
        />
        <View style={styles.container}>
        <View style={styles.buttonContainer}>
        <Button onPress={this._addToCalendar} title="Add to Calendar" />
        </View>
        <View style={styles.buttonContainer}>
        <Button onPress={this._deleteEvent} title="Remove Event" />
        </View>
        <View style={styles.buttonContainer}>
        <Button onPress={this._findEvent} title="Find Events" />
        </View>
        </View>
      </View>
    );
  }

  _findEvent = async () => {
    // TODO: Change which event is fetched maybe using a property passed from the CameraScreen using navigation properties

    // load the data for the event from the local storage
    _retrieveData("event").then(data => {
      // create the initial state
      const event = JSON.parse(data);
      const stateUpdate = {
        id: event.id,
        description: event.description,
        summary: event.summary,
        location: event.location,
        startTime: event.start.dateTime,
        endTime: event.end.dateTime
      };

      // set the initial state
      this.setState(stateUpdate);
    });
  };
  _deleteEvent = async () => {
    try {
      if (this.state.eventId) {
        const id = this.state.eventId;

        // TODO: Fix - Event is not deleted!
        await Calendar.deleteEventAsync(`${id}`, {
          instanceStartDate: "2019-02-19T15:00:00.000Z",
          futureEvents: true
        });

        this.setState({ eventId: null });
        Alert.alert(`Event with ID: ${id} has been deleted.`);
      }
    } catch (error) {
      console.error(error);
    }
  };

  _addToCalendar = async () => {
    //create dummy event
    let eventDetails = {
      title: "I am your new event",//
      startDate: "2019-02-19T15:00:00.000Z",///this.state.start, //got error saying saying it expected date to end in Z so edited: "2019-02-19T15:00:00.000Z+01:00",
      endDate: "2019-02-19T16:00:00.000Z",
      allDay: false,
      location: this.state.location,
      notes: "Testing add to Calendar",
      alarms: [
        {
          relativeOffset: -15,
          method: Calendar.AlarmMethod.DEFAULT
        }
      ],
      recurrenceRule: {
        frequency: Calendar.Frequency.DAILY,
        interval: 2,
        endDate: "2019-02-22T16:00:00.000Z",
        occurrence: 4
      },
      availability: "busy",
      timeZone: "GMT+1",
      url: "http://www..."
    };
    debugger;
    try {
      if (this.state.startTime){
        eventDetails.title= this.state.summary;
        eventDetails.location= this.state.location;
        eventDetails.startDate= new Date(this.state.startTime);
        eventDetails.endDate= new Date(this.state.endTime);
        eventDetails.recurrenceRule.endDate=new Date(this.state.endTime);
      }
      //add event to default calendar
      const eventId = await Calendar.createEventAsync(
          Calendar.DEFAULT,
          eventDetails
      );

      this.setState({ eventId: eventId });

      console.log("Event Id", eventId);

      Alert.alert(`The event ${eventId} is added to your calendar!`);
    } catch (error) {
      console.log("Error", error);
    }
  };


}

export default withNavigation(CalendarEventScreen);

const styles = StyleSheet.create({
  textContainer: {
    margin: 5,
    padding: 5
  }
});
