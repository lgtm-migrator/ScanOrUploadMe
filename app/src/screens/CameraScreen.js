import React from "react";

import {
  ActivityIndicator,
  Button,
  Clipboard,
  Image,
  Share,
  StatusBar,
  StyleSheet,
  Text,
  View,
  Alert
} from "react-native";
import { ImagePicker, Permissions, Calendar, Notifications } from "expo";
import { uploadImageAsync } from "../../api/uploadImage";
import { withNavigation } from "react-navigation";

import { _storeData, _retrieveData } from "../helpers/localStorage";

class CameraScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      image: null,
      uploading: false,
      calendarEvent: null
    };
  }

  static navigationOptions = {
    headerRight: (
      <Button
        onPress={() =>
          Alert.alert(
            "Dummy Button",
            "I should help to navigate to the created calendar event once an event has been created. Currently, I don't do much. The goal would be to only be active when the user has created an event (i.e. uploaded an image) and then to navigate back to the screen with the calendar event displayed."
          )
        }
        title="To Event"
      />
    )
  };

  async componentDidMount() {
    try {
      // get camera, camera roll and storage permission
      await Permissions.askAsync(Permissions.CAMERA);
      await Permissions.askAsync(Permissions.CAMERA_ROLL);
      await Permissions.askAsync(Permissions.CALENDAR);
      await Permissions.askAsync(Permissions.NOTIFICATIONS);
    } catch (error) {
      // display the error to the user
      Alert.alert(error);
      console.log({ error });
    }
  }

  render() {
    return (
      <View
        style={{
          flex: 1,
          alignItems: "center",
          justifyContent: "flex-end"
        }}
      >
        <StatusBar barStyle="default" />

        {this._maybeRenderImage()}
        {this._maybeRenderUploadingOverlay()}

        <View style={styles.textContainer}>
          <Text style={styles.mainText}>
            Choose how you are going to provide the input image.
          </Text>
        </View>

        <View style={styles.container}>
          <View style={styles.buttonContainer}>
            <Button onPress={this._pickImage} title="Select Image" />
          </View>
        <View style = {styles.headerLeft}>
        <Button onPress={this._addToCalendar} title="Add to Calendar" />
        </View>
          <View style={styles.buttonContainer}>
            <Button onPress={this._takePhoto} title="Take a photo" />
          </View>
          <View style={styles.buttonContainer}>
            <Button onPress={this._reset} title="Reset" />
          </View>
        </View>
      </View>
    );
  }

  _maybeRenderUploadingOverlay = () => {
    if (this.state.uploading) {
      return (
        <View
          style={[
            StyleSheet.absoluteFill,
            {
              backgroundColor: "rgba(0,0,0,0.4)",
              alignItems: "center",
              justifyContent: "center"
            }
          ]}
        >
          <ActivityIndicator color="#fff" animating size="large" />
        </View>
      );
    }
  };

  _maybeRenderImage = () => {
    const { image } = this.state;

    // if there is no image assigned yet don't render
    if (!image) {
      return;
    }

    return (
      <View style={styles.imageContainer}>
        <View
          style={{
            borderTopRightRadius: 3,
            borderTopLeftRadius: 3,
            overflow: "hidden"
          }}
        >
          <Image source={{ uri: image }} style={{ width: 300, height: 300 }} />
        </View>

        <Text
          onPress={this._copyToClipboard}
          onLongPress={this._share}
          style={{
            paddingVertical: 10,
            paddingHorizontal: 10
          }}
        >
          {image}
        </Text>
      </View>
    );
  };

  // share picture via social medias triggered by long press on image URL
  _share = () => {
    Share.share({
      message: `Check out this photo: ${this.state.image}`,
      title: "Check out this photo!",
      url: this.state.image
    });
  };

  // copy link URL triggered by press on image URL
  _copyToClipboard = () => {
    Clipboard.setString(this.state.image);
    Alert.alert("Copied URL to clipboard");
  };

  // remove currently display image
  _reset = () => {
    this.setState({ image: null, calendarEvent: null });
  };

  // take photo with the phone camera
  _takePhoto = async () => {
    const pickerResult = await ImagePicker.launchCameraAsync({
      allowsEditing: true,
      quality: 1
    });

    this._handleImagePicked(pickerResult);
  };


  



  // choose photo from camera roll
  _pickImage = async () => {
    const pickerResult = await ImagePicker.launchImageLibraryAsync({
      allowsEditing: true,
      quality: 1
    });

    this._handleImagePicked(pickerResult);
  };

  // handle the chosen image
  _handleImagePicked = async pickerResult => {
    let uploadResponse;
    let uploadResult;
    let responseJson;

    try {
      this.setState({ uploading: true });

      if (!pickerResult.cancelled) {
        // upload the image to server
        uploadResponse = await uploadImageAsync(pickerResult.uri);
        uploadResult = await uploadResponse.json();

        if (!uploadResult.uploaded) {
          console.log("Uploaded to server failed.");
          this.setState({
            image: null,
            calendarEvent: null
          });
          Alert.alert("Image Upload Failed!", "Uploaded to server failed.");
        } else if (!uploadResult.success) {
          console.log(
            "Uploaded to server successfull but no result from Google Cloud."
          );
          this.setState({
            image: null,
            calendarEvent: null
          });
          Alert.alert(
            "No Feedback from Google Cloud",
            "Uploaded to server successfull but no result from Google Cloud."
          );
        } else {
          // Upload was sucessfull and got result from Google Cloud
          this.setState({
            // set state to URL of upload location
            image: uploadResult.location,
            // set process result of Google Cloud
            calendarEvent: uploadResult
          });

          console.log(this.state.calendarEvent);
          // Currently stores the received event locally
          // TODO: decide what we store and how
          // TODO: forwarding the key: "event" to the Calendar screen
          _storeData("event", uploadResult.calendarEvent).then(
            // navigate to the new screen as last action
            this.props.navigation.navigate("Calendar", {
              photoUri: uploadResult.location,
              eventName: "event"
            })
          );
        }
      }
    } catch (error) {
      console.log({ uploadResponse });
      console.log({ uploadResult });
      console.log({ responseJson });
      console.log({ error });

      Alert.alert("Upload Failed!", error.message);
    } finally {
      this.setState({ uploading: false });
    }
  };
}

export default withNavigation(CameraScreen);

const styles = StyleSheet.create({
  imageContainer: {
    marginTop: 10,
    marginBottom: 10,
    width: 300,
    borderRadius: 3,
    elevation: 2,
    shadowColor: "rgba(0,0,0,1)",
    shadowOpacity: 0.2,
    shadowOffset: { width: 4, height: 4 },
    shadowRadius: 5
  },
  container: {
    alignItems: "center",
    flexDirection: "row",
    marginBottom: 20
  },
  buttonContainer: {
    marginHorizontal: 5
  },
  textContainer: {
    alignItems: "center",
    marginHorizontal: 10,
    marginBottom: 10
  },
  mainText: {
    fontSize: 16,
    color: "rgba(96,100,109, 1)",
    lineHeight: 24,
    textAlign: "center"
  }
});
