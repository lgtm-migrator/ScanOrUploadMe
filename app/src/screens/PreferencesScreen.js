import React from "react";
import {
  Text,
  View,
  StyleSheet,
  Picker,
  Platform,
  Alert,
  Button
} from "react-native";
import { withNavigation } from "react-navigation";
import { Calendar, Permissions } from "expo";

import { testBackendAPI, getAccessToken } from "../../api/authorise";

// redux
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { setDefaultCalendar } from "../actions/calendarActions";
import { setDefaultScanLanguage } from "../actions/settingsActions";
import { setAccessToken, clearAccessToken } from "../actions/securityActions";

class PreferencesScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      authorizationMessage: null,
      calendarEvent: null,
      calendars: [],
      data: [
        { name: "English", value: "EN", id: 1 },
        { name: "Swedish", value: "SE", id: 2 }
      ]
    };
  }

  componentWillReceiveProps() {
    console.log("component received props:", this.props.security);
  }

  async componentDidMount() {
    // check the necessary permissions
    this._checkPermissions()
      .then(permission => {
        if (permission) {
          // retrieve all local calendars which are modifiable
          this._findCalendars();
        } else {
          Alert.alert(
            "Permission Error",
            "We could not retrieve the local calendars because we're missing the necessary permissions to access the local calendars."
          );
        }
      })
      .catch(error => {
        console.log(error);
      });
  }

  _checkPermissions = async () => {
    let permission = false;

    // check the calendar permission
    const calendarResponse = await Permissions.askAsync(Permissions.CALENDAR);
    permission = calendarResponse.status === "granted";

    // check the reminder calendars on ios
    if (Platform.OS === "ios") {
      const reminderResponse = await Permissions.askAsync(
        Permissions.REMINDERS
      );
      permission = reminderResponse.status === "granted";
    }

    return permission;
  };

  _findCalendars = async () => {
    // retrieve all event calendars
    const eventCalendars = await Calendar.getCalendarsAsync("event");

    // on ios, also retrieve all reminder calendars
    const reminderCalendars =
      Platform.OS === "ios" ? await Calendar.getCalendarsAsync("reminder") : [];
    const calendars = [...eventCalendars, ...reminderCalendars];

    // filter out non-modifiable calendars
    const modifiableCalendars = calendars.filter(
      calendar => calendar.allowsModifications
    );

    this.setState({ calendars: modifiableCalendars });
  };

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.defaultSettingContainer}>
          <Text style={styles.textContainer}>
            ID of the selected Calendar ID:{" "}
            {this.props.calendar.activeCalendarId}
          </Text>

          <Picker
            style={styles.pickerContainer}
            selectedValue={this.props.calendar.activeCalendarId}
            onValueChange={itemValue =>
              this.props.setDefaultCalendar(itemValue)
            }
          >
            {this.state.calendars.length ? (
              this.state.calendars.map(calendar => (
                <Picker.Item
                  label={calendar.title}
                  value={calendar.id}
                  key={calendar.id}
                />
              ))
            ) : (
              <Picker.Item
                label="Error: No Permission to access the calendars"
                value={1}
              />
            )}
          </Picker>
        </View>

        <View style={styles.defaultSettingContainer}>
          <Text style={styles.textContainer}>
            Default Scan Language: {this.props.settings.scanLanguage}
          </Text>

          <Picker
            style={styles.pickerContainer}
            selectedValue={this.props.settings.scanLanguage}
            onValueChange={itemValue => {
              this.props.setDefaultScanLanguage(itemValue);
            }}
          >
            <Picker.Item label="English" value={"EN"} />
            <Picker.Item label="Swedish" value={"SE"} />
            <Picker.Item label="German" value={"DE"} />
          </Picker>
        </View>

        <View style={styles.defaultSettingContainer}>
          <Text style={styles.textContainer}>
            Authorization: Authorize yourself in order to use our backend
            services!
          </Text>
          <Button title={"Authorize"} onPress={this._authorizeAPI} />
          <Text>Access Token: {this.props.security.accessToken}</Text>
          <Button title={"Test Backend"} onPress={this._testAPI} />
          <Text>Authorization Message: {this.state.authorizationMessage}</Text>
          <Button title={"Clear Access Token"} onPress={this._clearToken} />
        </View>
      </View>
    );
  }

  _authorizeAPI = async () => {
    const accessToken = await getAccessToken();
    this.props.setAccessToken(accessToken);
  };

  _clearToken = () => {
    if (this.props.security.accessToken) {
      this.props.clearAccessToken();
      this.setState({ authorizationMessage: null });
    }
  };

  _testAPI = async () => {
    if (!this.props.security.accessToken) {
      Alert.alert("No access token - authorize first!");
      return;
    }
    const reply = await testBackendAPI(this.props.security.accessToken);
    this.setState({ authorizationMessage: reply._bodyText });
  };
}

const mapStateToProps = state => {
  const { calendar, security, settings } = state;
  return { calendar, security, settings };
};

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      setDefaultCalendar,
      setDefaultScanLanguage,
      setAccessToken,
      clearAccessToken
    },
    dispatch
  );

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withNavigation(PreferencesScreen));

const styles = StyleSheet.create({
  container: {
    alignItems: "flex-start",
    flexDirection: "column"
  },
  defaultSettingContainer: {
    display: "flex",
    flexDirection: "column",
    alignItems: "flex-start",
    borderRadius: 4,
    borderWidth: 0.5,
    borderColor: "#d6d7da",
    padding: 5,
    marginTop: 5,
    width: "100%"
  },
  pickerContainer: {
    height: 50,
    width: 150
  },
  subContainer: {
    margin: 8
  },
  dropDownContainer: {
    borderWidth: 0.5,
    borderRadius: 4,
    padding: 8
  },
  dropDownText: {
    fontSize: 20,
    margin: 8
  }
});
