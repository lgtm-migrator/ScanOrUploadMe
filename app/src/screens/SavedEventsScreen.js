import React from 'react';
import { AsyncStorage, Button, Text,
     FlatList, ListView, TextInput,
    TouchableHighlight,
    View, StyleSheet, Alert } from 'react-native';
//import styled from 'styled-components';
import "../helpers/eventsList";



let now = new Date();
let start = new Date(now.setHours(now.getHours() + 1)).toString();
let end = new Date(now.setHours(now.getHours() + 2)).toString();

let calendarEvent = {
    description: 'Guava Peeling Party',
    id: 1234,
    picUri:"https://t2.rbxcdn.com/34e222a0bf98d0a46de65de2ece139eb",
    summary:
        'Hi John\nI would like to invite you to my birthday party!\nBest James',
    location: 'Klostergatan, Uppsala, Sweden',
    startTime: start,
    endTime: end,
    reminders: {
        overrides: [{ method: 'popup', minutes: 15 }],
        useDefault: false
    },
    updated: Date()
};

let calendarEvent2 = {
    description: 'Puppy Petting Party',
    id: 5555,
    picUri:"https://t2.rbxcdn.com/34e222a0bf98d0a46de65de2ece139eb",
    summary:
        'Hi John\nI would like to invite you to my Puppy Petting Party!\nBest James',
    location: 'Klostergatan, Uppsala, Sweden',
    startTime: start,
    endTime: end,
    reminders: {
        overrides: [{ method: 'popup', minutes: 15 }],
        useDefault: false
    },
    updated: Date()
};
let calendarEvent3 = {
    description: 'Confetti Party',
    id: 5050,
    picUri:"https://t2.rbxcdn.com/34e222a0bf98d0a46de65de2ece139eb",
    summary:
        'Hi ho, hi,ho \nI would like to invite you to my Puppy Petting Party!\nBest James',
    location: 'Klostergatan, Uppsala, Sweden',
    startTime: start,
    endTime: end,
    reminders: {
        overrides: [{ method: 'popup', minutes: 15 }],
        useDefault: false
    },
    updated: Date()
};

export default class SavedEventsScreen extends React.Component {

    _onPressButton= () => {
        Alert.alert('Event details to come');
        //this.props.onPressItem(this.props.id)
    };
    static navigationOptions = {
        title: 'Saved Events',
    };
    constructor(props) {
        super(props);
        this.state = {
            id: "",
            summary: "",
            description: "",
            location: "",
            startTime: "",
            endTime: ""
        };



        const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
        this.state = {
            dataSource: ds.cloneWithRows([calendarEvent.description, calendarEvent2.description, calendarEvent.description]),
        };

    }

    render() {
        return (
            <View style={styles.container} >

                <FlatList
                    data={[calendarEvent2,calendarEvent,calendarEvent3]}//,calendarEvent2,calendarEvent,calendarEvent3,calendarEvent2,calendarEvent,calendarEvent3
                    showsVerticalScrollIndicator={false}
                    renderItem={({item}) =>
                        <TouchableHighlight onPress={this._onPressButton} underlayColor="white">
                        <View style={styles.flatview}>

                            <Text style={styles.name}>{item.description}</Text>
                            <Text style={styles.brief}>{item.startTime}</Text>

                        </View>
                        </TouchableHighlight>
                    }
                    keyExtractor={item => item.summary}
                />
            </View>



    );
    }
    //USE code in comment below to add button to retrieve local data
    /*<View style={styles.container}>
                    <View style={styles.buttonContainer}>
                        <Button onPress={this._retrieveEvent} title="Retrieve" />
                    </View>
                </View>*/

    _retrieveEvent = async () =>{
        try { //userId
            let key='userId';
            const value = await AsyncStorage.getItem(key);
            console.log('RETRIEVED: '+value);
            return value;
        } catch (error) {
            // Error retrieving data
            console.log({ error });
            console.error(error);
        }
    }

}

const styles = StyleSheet.create({
    buttonContainer: {
        marginHorizontal: 5,
        padding: 20,
    },
    container: {
        backgroundColor: '#fff',
        flex: 1,
        marginTop: 20,
        padding: 20,
        paddingTop: 20,
    },
    photo: {
        borderRadius: 20,
        height: 40,
        width: 40,

    },
    text: {
        fontSize: 16,
        marginLeft: 12,

    },
    h2text: {
        marginTop: 10,
        fontFamily: 'Helvetica',
        fontSize: 28,
        fontWeight: 'bold',
    },
    flatview: {
        justifyContent: 'center',
        paddingTop: 30,
        borderRadius: 2
        //backgroundColor:'blue'
    },
    name: {
        fontFamily: 'Verdana',
        fontSize: 18
    },
    brief: {
        color: 'gray'
    }

});
