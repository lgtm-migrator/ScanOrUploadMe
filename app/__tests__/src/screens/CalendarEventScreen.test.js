import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

import { CalendarEventScreen } from "../../../src/screens/CalendarEventScreen";

import React from "react";

import renderer from "react-test-renderer";

describe("CalendarEventScreen: Tests", () => {
  const initialEvents = {
    currentEventId: null,
    currentEvent: null,
    events: []
  };

  const someEvents = {
    currentEventId: 1,
    currentEvent: {
      id: 1,
      description: "Test",
      summary: "Test",
      location: "Test",
      startTime: "Test",
      endTime: "Test"
    },
    events: [
      {
        id: 1,
        description: "Test",
        summary: "Test",
        location: "Test",
        startTime: "Test",
        endTime: "Test"
      }
    ]
  };

  const dummyState = {
    id: 12312312,
    description: "DummyTitle",
    summary: "DummyText",
    location: "DummyLocation",
    startTime: "2019-02-27T16:00:00.000Z",
    endTime: "2019-02-27T18:00:00.000Z"
  };

  const initializedState = {
    id: 1,
    description: "Test",
    summary: "Test",
    location: "Test",
    startTime: "Test",
    endTime: "Test"
  };

  jest.useFakeTimers();

  beforeEach(() => {});

  describe("CalendarEventScreen: Shallow Render Component", () => {
    it("renders the component without any real events", () => {
      const wrapper = shallow(<CalendarEventScreen events={initialEvents} />);
      const component = wrapper.instance();

      expect(component.state).toEqual(dummyState);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders the component with a real event", () => {
      const wrapper = shallow(<CalendarEventScreen events={someEvents} />);
      const component = wrapper.instance();

      expect(component.state).toEqual(initializedState);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe("test TextInput changes", () => {
    it("test the _updateSingleStateValue function", () => {
      const instanceOf = renderer
        .create(<CalendarEventScreen events={initialEvents} />, {
          disableLifecycleMethods: true
        })
        .getInstance();

      // call the state update method
      instanceOf._updateSingleStateValue("description", "TEST_TEST");

      // check that the update value is correct
      expect(instanceOf.state.description).toEqual("TEST_TEST");
    });
  });
});
