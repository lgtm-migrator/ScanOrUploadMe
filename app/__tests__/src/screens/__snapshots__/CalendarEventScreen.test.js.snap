// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarEventScreen: Tests CalendarEventScreen: Shallow Render Component renders the component with a real event 1`] = `
<ScrollViewMock>
  <Component
    style={
      Object {
        "backgroundColor": "#fff",
        "justifyContent": "center",
        "margin": 5,
        "padding": 5,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      multiline={true}
      onChangeText={[Function]}
      placeholder="SUMMARY"
      style={
        Object {
          "height": 40,
        }
      }
      underlineColorAndroid="transparent"
      value="Test"
    />
    <TextInput
      allowFontScaling={true}
      multiline={true}
      onChangeText={[Function]}
      placeholder="DESCRIPTION"
      style={
        Object {
          "height": 40,
        }
      }
      underlineColorAndroid="transparent"
      value="Test"
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="LOCATION:"
      style={
        Object {
          "height": 40,
        }
      }
      underlineColorAndroid="transparent"
      value="Test"
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="START_TIME"
      style={
        Object {
          "height": 40,
        }
      }
      underlineColorAndroid="transparent"
      value="Test"
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="END_TIME"
      style={
        Object {
          "height": 40,
        }
      }
      underlineColorAndroid="transparent"
      value="Test"
    />
  </Component>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 20,
      }
    }
  >
    <Component
      style={
        Object {
          "marginHorizontal": 5,
          "marginTop": 50,
        }
      }
    >
      <Button
        onPress={[Function]}
        title="Add to Calendar"
      />
    </Component>
    <Component
      style={
        Object {
          "marginHorizontal": 5,
          "marginTop": 50,
        }
      }
    >
      <Button
        onPress={[Function]}
        title="Remove Event"
      />
    </Component>
    <Component
      style={
        Object {
          "marginHorizontal": 5,
          "marginTop": 50,
        }
      }
    >
      <Button
        onPress={[Function]}
        title="Update Event"
      />
    </Component>
  </Component>
</ScrollViewMock>
`;

exports[`CalendarEventScreen: Tests CalendarEventScreen: Shallow Render Component renders the component without any real events 1`] = `
<ScrollViewMock>
  <Component
    style={
      Object {
        "backgroundColor": "#fff",
        "justifyContent": "center",
        "margin": 5,
        "padding": 5,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      multiline={true}
      onChangeText={[Function]}
      placeholder="SUMMARY"
      style={
        Object {
          "height": 40,
        }
      }
      underlineColorAndroid="transparent"
      value="DummyText"
    />
    <TextInput
      allowFontScaling={true}
      multiline={true}
      onChangeText={[Function]}
      placeholder="DESCRIPTION"
      style={
        Object {
          "height": 40,
        }
      }
      underlineColorAndroid="transparent"
      value="DummyTitle"
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="LOCATION:"
      style={
        Object {
          "height": 40,
        }
      }
      underlineColorAndroid="transparent"
      value="DummyLocation"
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="START_TIME"
      style={
        Object {
          "height": 40,
        }
      }
      underlineColorAndroid="transparent"
      value="2019-02-27T16:00:00.000Z"
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="END_TIME"
      style={
        Object {
          "height": 40,
        }
      }
      underlineColorAndroid="transparent"
      value="2019-02-27T18:00:00.000Z"
    />
  </Component>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 20,
      }
    }
  >
    <Component
      style={
        Object {
          "marginHorizontal": 5,
          "marginTop": 50,
        }
      }
    >
      <Button
        onPress={[Function]}
        title="Add to Calendar"
      />
    </Component>
    <Component
      style={
        Object {
          "marginHorizontal": 5,
          "marginTop": 50,
        }
      }
    >
      <Button
        onPress={[Function]}
        title="Remove Event"
      />
    </Component>
    <Component
      style={
        Object {
          "marginHorizontal": 5,
          "marginTop": 50,
        }
      }
    >
      <Button
        onPress={[Function]}
        title="Update Event"
      />
    </Component>
  </Component>
</ScrollViewMock>
`;
